analyzers/nodejs/nodejs_test.go

```diff
@@ -13,6 +13,58 @@ import (
 	"github.com/fossas/fossa-cli/pkg"
 )
 
+/*
+   └─┬ a@1.0.0
+     ├─┬ b@2.0.0
+     │ ├── c@3.0.0
+     │ └── d@4.0.0
+     └── c@3.0.0
+*/
+
+// TestNDepsTransitiveImports verifies that each dependency returned by Analyze()
+// in the transitive dependency list contains the correct dependency imports.
+func TestNDepsTransitiveImports(t *testing.T) {
+	m := module.Module{
+		Dir:  filepath.Join("testdata", "transitive-deps"),
+		Type: pkg.NodeJS,
+	}
+
+	a, err := analyzers.New(m)
+	assert.NoError(t, err)
+
+	a.(*nodejs.Analyzer).NPM = MockNPM{
+		JSONFilename: filepath.Join("testdata", "transitive-deps", "npm-ls-json.json"),
+	}
+
+	deps, err := a.Analyze()
+	assert.NoError(t, err)
+
+	assert.Equal(t, 1, len(deps.Direct))
+	assertImport(t, deps.Direct, "a", "1.0.0")
+
+	assert.Equal(t, 4, len(deps.Transitive))
+
+	packageA := assertPackageReturnImports(deps.Transitive, "a", "1.0.0")
+	assert.NotEqual(t, packageA, pkg.Package{})
```

Comment:
Here and below, I believe you can also use `assert.NotZero`.

Reply:
You're right, I need to read up on the available assertions